state/factory/workingset.go

```diff
@@ -195,12 +199,32 @@ func (ws *workingSet) runAction(
 			)
 		}
 		if receipt != nil {
+			isBlobTx := false // TODO: get blob type from action
+			if protocol.MustGetFeatureCtx(ctx).EnableBlobTransaction && isBlobTx {
+				if err = ws.handleBlob(ctx, selp, receipt); err != nil {
+					return nil, err
+				}
+			}
 			return receipt, nil
 		}
 	}
 	return nil, errors.New("receipt is empty")
 }
 
+func (ws *workingSet) handleBlob(ctx context.Context, act *action.SealedEnvelope, receipt *action.Receipt) error {
+	// Deposit blob fee
+	blobNum := 0 // TODO: get blob number from action
+	receipt.BlobGasUsed = uint64(blobNum) * params.BlobTxBlobGasPerBlob
+	receipt.BlobGasPrice = block.CalcBlobFee(protocol.MustGetBlockchainCtx(ctx).Tip.ExcessBlobGas)
+	blobFee := new(big.Int).Mul(receipt.BlobGasPrice, new(big.Int).SetUint64(receipt.BlobGasUsed))
+	logs, err := rewarding.DepositGas(ctx, ws, nil, protocol.BurnGasOption(blobFee, iotextypes.TransactionLogType_BLOB_FEE))
```

Comment:
add a test to verify balance is correct
`initBalance = remaining + tip + base fee + blob base fee`